clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/KafkaInputDictTrainer.java

```diff
@@ -207,14 +207,15 @@ synchronized byte[] trainDict(Optional<PubSubConsumerAdapter> reusedConsumerOpti
 
     // Reuse the same Kafka Consumer across all partitions avoid log flooding
     PubSubConsumerAdapter reusedConsumer = reusedConsumerOptional.orElseGet(
-        () -> new ApacheKafkaConsumerAdapterFactory().create(
-            KafkaInputUtils.getConsumerProperties(jobConf),
-            false,
-            new PubSubMessageDeserializer(
-                KafkaInputUtils.getKafkaValueSerializer(jobConf),
-                new LandFillObjectPool<>(KafkaMessageEnvelope::new),
-                new LandFillObjectPool<>(KafkaMessageEnvelope::new)),
-            null));
+        () -> PubSubClientsFactory.createConsumerFactory(KafkaInputUtils.getConsumerProperties(jobConf))
```

Comment:
Can we compute the properties once and reuse them, instead of calling `KafkaInputUtils.getConsumerProperties(jobConf)` twice before passing them to `PubSubClientsFactory.createConsumerFactory`?